### PR TITLE
Jinja2.py: Add support for ignore comments

### PR DESCRIPTION
**Pull request by sks444**
Closes https://github.com/coala/coala/issues/5107

**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [x] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [x] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `corobo mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by sks444** [comment]
Yeah, I also think we can add more extensions like: `.j2`,`.jinja`, and `.jinja2`. @jayvdb, Is it okay to add these to extensions?

**Comment by jayvdb** [comment]
ya all those are appropriate additions.

**Comment by sks444** [comment]
Actually, I am a little bit confused about adding tests. I don't see test cases explicitly, for any of the language definitions [here](https://github.com/coala/coala/tree/master/coalib/bearlib/languages/definitions).
E.g.: see this [PR](https://github.com/coala/coala/commit/d87aa4a7cc88bedd72d5231f109a1c0a6a456032) by Nalin, he added a new language definition but not the tests or this [PR](https://github.com/coala/coala/commit/6960d2fe95d0ff44dd1ac71d4261a21e765a7c8f#diff-f73125f50e36e1a480ec971effe75aeb) by sils in which he added many new language definitions but not tests for any of them, explicitly. I think tests for all the language definitions are covered [here](https://github.com/coala/coala/blob/6960d2fe95d0ff44dd1ac71d4261a21e765a7c8f/tests/bearlib/languages/LanguageDefinitionTest.py)? Please correct me if I am wrong.

**Comment by jayvdb** [comment]
You are adding support for ignore comments.  The tests are in the ignore comments tests.

**Comment by kriti21** [comment]
https://github.com/coala/coala/tree/master/tests/bearlib/languages/documentation This might be of some help.

File: tests/results/result_actions/IgnoreResultActionTest.py [hunk 1]
```diff
@@ -101,6 +101,16 @@ def test_ignore(self):
             with open(f_a, 'r') as f:
                 self.assertEqual(file_diff_dict[f_a].modified, f.readlines())
 
+            # Apply a third patch, old patch has to stay!
+            uut.apply(Result.from_values('bear', 'msg', f_a, 3),
+                      file_dict, file_diff_dict, 'jinja2')
+            self.assertEqual(
+                file_diff_dict[f_a].modified,
+                ['1  /* Ignore else */\n', '2  // Ignore origin\n',
```
Comment (jayvdb) [inline tests/results/result_actions/IgnoreResultActionTest.py]: these comments are obviously invalid for jinja, so it does not make sense to add your test to this method.  Create a new test method for jinja.

Comment (RaiVaibhav) [reply tests/results/result_actions/IgnoreResultActionTest.py]: You can use simply use [get_ignore_comment](https://github.com/coala/coala/blob/3c3fab54063f05960a3862ea3b2c83174341a2cf/coalib/results/result_actions/IgnoreResultAction.py#L81) to directly check whether your definition is working or not and accordingly write a test for that.

**Comment by jayvdb** [comment]
ack https://github.com/coala/coala/pull/5428/commits/8dfe8376e70b94d3a6f359dc656575b9fb42b35f

**Comment by jayvdb** [comment]
@gitmate-bot ff

**Pull request merged**